mysys/CMakeLists.txt

```diff
@@ -92,6 +92,12 @@ ELSEIF(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|AARCH64")
   IF(CMAKE_COMPILER_IS_GNUCC)
     include(CheckCXXSourceCompiles)
 
+    CHECK_CXX_SOURCE_COMPILES("
```

Comment:
Please use CHECK_INCLUDE_FILES https://cmake.org/cmake/help/latest/module/CheckIncludeFiles.html

Reply:
Its checking for a compile error in the header. `check_symbol_exists(__ARM_FP arm_neon.h HAVE_ARMV8_NEON_SOFTFP)` maybe.

Reply:
CHECK_INCLUDE_FILES checks for compile error, same thing

Reply:
Now it use that approach and it looks lot cleaner. Although people seems to hate  `CHECK_INCLUDE_FILES

Reply:
I've tested this on Debian Salsa also and seems to work as expected. With or without armel

Reply:
I do not know why people would hate `CHECK_INCLUDE_FILES`. Rather, a raw `CHECK_SOURCE_COMPILES` with an empty  main() usually a sign of misunderstanding. While it might be used for either a check that is too complex, or a test for compiler bug, but it is very seldom. Much more often, compact, and task-specific CHECK_INCLUDE_FILES, CHECK_FUNCTION_EXISTS, CHECK_SYMBOL_EXISTS would do the job.